server/app/views/admin/programs/ProgramBlockBaseView.java

```diff
@@ -0,0 +1,108 @@
+    DivTag description =
+        div(programDefinition.localizedDescription().getDefault()).withClasses("text-sm");
+    DivTag adminNote =
+        div()
+            .withClasses("text-sm")
+            .with(span("Admin note: ").withClasses("font-semibold"))
+            .with(span(programDefinition.adminDescription()));
+
+    ButtonTag editDetailsButton =
+        ViewUtils.makeSvgTextButton(getEditButtonText(), Icons.EDIT)
+            .withClasses(AdminStyles.SECONDARY_BUTTON_STYLES, "my-5");
+    asRedirectElement(editDetailsButton, getEditButtonUrl(programDefinition));
+
+    return div(
+            ViewUtils.makeBadge(getProgramDisplayStatus()),
+            title,
+            description,
+            adminNote,
+            editDetailsButton)
+        .withClasses("bg-gray-100", "text-gray-800", "shadow-md", "p-8", "pt-4", "-mx-2");
+  }
+
+  /** Renders a div presenting the predicate definition for the admin. */
+  protected final DivTag renderExistingPredicate(
+      String blockName,
+      PredicateDefinition predicateDefinition,
+      ImmutableList<QuestionDefinition> questionDefinitions) {
+    DivTag container = div();
+
+    if (predicateDefinition
+        .computePredicateFormat()
+        .equals(PredicateDefinition.PredicateFormat.SINGLE_QUESTION)) {
+      return container.with(
+          text(predicateDefinition.toDisplayString(blockName, questionDefinitions)));
+    } else if (!predicateDefinition
+        .computePredicateFormat()
+        .equals(PredicateDefinition.PredicateFormat.OR_OF_SINGLE_LAYER_ANDS)) {
+      throw new IllegalArgumentException(
+          String.format(
+              "Predicate type %s is unsupported.", predicateDefinition.computePredicateFormat()));
+    }
+
+    ImmutableList<PredicateExpressionNode> andNodes =
+        predicateDefinition.rootNode().getOrNode().children();
+
+    if (andNodes.size() == 1) {
+      return container.with(
+          text(
+              blockName
+                  + " is "
+                  + predicateDefinition.action().toDisplayString()
+                  + " "
+                  + andNodes.get(0).getAndNode().toDisplayString(questionDefinitions)));
+    }
+
+    container.with(
+        text(blockName + " is " + predicateDefinition.action().toDisplayString() + " any of:"));
+    UlTag conditionList = ul().withClasses("list-disc", "ml-4", "mb-4");
+
+    andNodes.stream()
+        .map(PredicateExpressionNode::getAndNode)
+        .forEach(andNode -> conditionList.with(li(andNode.toDisplayString(questionDefinitions))));
+
+    return container.with(conditionList);
+  }
+
+  /** Returns the string that will be shown on the Edit button */
+  protected abstract String getEditButtonText();
+
+  /** Returns the navigation destination for the Edit button */
+  protected abstract String getEditButtonUrl(ProgramDefinition programDefinition);
+
+  /**
+   * Returns the badge which represents the status of the program. It will be shown at the top of
```

Comment:
**nitpick (blocking):** Update to rephrase "badge" since you renamed it.

Reply:
Done